tiledb/sm/c_api/tiledb.cc

```diff
@@ -1207,6 +1214,61 @@ int32_t tiledb_ctx_alloc(tiledb_config_t* config, tiledb_ctx_t** ctx) try {
   return TILEDB_ERR;
 }
 
+int32_t tiledb_ctx_alloc_with_error(
+    tiledb_config_t* config, tiledb_ctx_t** ctx, tiledb_error_t** error) try {
+  if (config != nullptr && config->config_ == nullptr)
+    return TILEDB_ERR;
+
+  // Create a context object
+  *ctx = new (std::nothrow) tiledb_ctx_t;
+  if (*ctx == nullptr)
+    return TILEDB_OOM;
+
+  // Create a context object
+  (*ctx)->ctx_ = new (std::nothrow) tiledb::sm::Context();
```

Comment:
`nothrow` guarantees that this call returns `nullptr` if the constructor fails. However, looking at `Context::Context()` it's not clear to me that the _only_ reason for a failure would be a badalloc. The `Status` constructor looks safe, but the `Logger` constructor initializes spdlog state which seems like it could potentially throw for non-allocation reasons (have not audited).

If we allow this call to throw, we have catch patterns below to return the exception message.